experimental/emulated_vault/opt/emulated-vault/adapter/fs.py

```diff
@@ -0,0 +1,313 @@
+#
```

Comment:
The current score of this file is -7.37/10, but that can be raised as high as 6.24 just by using tabs for indentation.

By the way, for future work into making more adapters I just wanna suggest writing out an [abstract base class](https://docs.python.org/3/library/abc.html) for people to implement. Seems like a good use of it, and it'd be helpful for type hints if you ever move to that.

Reply:
I actually started to work on it this morning but did not wanted to move files in the middle of the review.
Would you prefer me to push it now?

Reply:
You mean abstract base classes? I don't mind if you want to push updates, I can start my review over. If you'd rather see this merged first and then work on top of that, that's fine too.

Reply:
Not exactly abstract, but base with some un-implemented functions.
See pushed update